### PR TITLE
Damage on Contact with Terrain 

### PR DESCRIPTION
**Pull request by KishanBapodra**
Is what I did right not too sure 
I added code for listening to game over event in main.js need to figure out how to remove the player though

### CODE REVIEW

**Comment by FarazzShaikh** [comment]
Make the changes and then you can work on the damage system that will integrate with the shooting system

**Review by FarazzShaikh — CHANGES_REQUESTED** [review]

File: server/src/main.js [hunk 1]
```diff
@@ -36,6 +36,11 @@ export default function main(http) {
 			game.movePlayer(socket, typeOfMovement)
 		);
 
+		// Listens for Game_Over event
+		socket.on(EVENTS.GAME_OVER, () => 
+			game.removePlayer(socket)
+		);
+
```
[the remark this answers is not in the document]
Comment (KishanBapodra) [reply server/src/main.js]: Alright so its like if didcollide is true then its emitting the server hit ground.req and that is acknowledged in the client main.js?

[the remark this answers is not in the document]
Comment (KishanBapodra) [reply server/src/main.js]: or is it the other way around as in when there is a collision its acknowledged i.e. emit(REQUE.SER.HI.GROUN.ACK) and on the client side in main it listens for the server,hit,gro.req

[the remark this answers is not in the document]
Comment (FarazzShaikh) [reply server/src/main.js]: The server will emit `EVENT.SERVER_PLAYER_HIT_GROUND.req` the client will listen for this event and then respond with an acknowledgement -  `EVENT.SERVER_PLAYER_HIT_GROUND.ack`. Then the server will listen for this acknowledgement and then delete the player from the server and send an updated player list to everyone else by emitting `EVENTS.SERVER_UPDATE_PLAYERS`. The other players will listen for this and update their player lists accordingly.

**Review by FarazzShaikh — APPROVED** [review]

**Pull request merged**